pkg/kademlia/routinggraph/routinggraph.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package routinggraph
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/pb"
+)
+
+func extendPrefix(prefix string, bit bool) string {
+	if bit {
+		return prefix + "1"
+	}
+	return prefix + "0"
+}
+
+// Draw writes the routing graph obtained using a GetBucketListResponse in the specified file
+func Draw(file *os.File, info *pb.GetBucketListResponse) (err error) {
+	_, err = file.Write([]byte("digraph{\nnode [shape=plaintext, fontname=\"Courier\"];edge [dir=none];rankdir=LR;\n"))
+	if err != nil {
+		return err
+	}
+	defer func() {
+		_, errWrite := file.Write([]byte("}\n"))
+		err = errs.Combine(err, errWrite)
+		err = errs.Combine(err, file.Close())
```

Comment:
Don't move the closing responsibility, since it can cause bugs. Keep the closing of something close to where it was opened/created.

Currently this would end up closing stdout.